src/main/java/org/alfresco/repo/blog/BlogServiceImpl.java

```diff
@@ -0,0 +1,633 @@
+import org.alfresco.query.PagingRequest;
+import org.alfresco.query.PagingResults;
+import org.alfresco.repo.blog.cannedqueries.BlogEntity;
+import org.alfresco.repo.blog.cannedqueries.DraftsAndPublishedBlogPostsCannedQuery;
+import org.alfresco.repo.blog.cannedqueries.DraftsAndPublishedBlogPostsCannedQueryFactory;
+import org.alfresco.repo.blog.cannedqueries.GetBlogPostsCannedQuery;
+import org.alfresco.repo.blog.cannedqueries.GetBlogPostsCannedQueryFactory;
+import org.alfresco.repo.content.MimetypeMap;
+import org.alfresco.repo.search.impl.lucene.LuceneUtils;
+import org.alfresco.repo.security.authentication.AuthenticationUtil;
+import org.alfresco.repo.security.permissions.AccessDeniedException;
+import org.alfresco.repo.site.SiteServiceImpl;
+import org.alfresco.service.cmr.blog.BlogPostInfo;
+import org.alfresco.service.cmr.blog.BlogService;
+import org.alfresco.service.cmr.dictionary.DataTypeDefinition;
+import org.alfresco.service.cmr.dictionary.DictionaryService;
+import org.alfresco.service.cmr.repository.ChildAssociationRef;
+import org.alfresco.service.cmr.repository.ContentService;
+import org.alfresco.service.cmr.repository.ContentWriter;
+import org.alfresco.service.cmr.repository.DuplicateChildNodeNameException;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.NodeService;
+import org.alfresco.service.cmr.search.ResultSet;
+import org.alfresco.service.cmr.search.SearchParameters;
+import org.alfresco.service.cmr.search.SearchService;
+import org.alfresco.service.cmr.security.PermissionService;
+import org.alfresco.service.cmr.site.SiteService;
+import org.alfresco.service.cmr.tagging.TaggingService;
+import org.alfresco.service.namespace.NamespaceService;
+import org.alfresco.service.namespace.QName;
+import org.alfresco.service.transaction.TransactionService;
+import org.alfresco.util.ISO9075;
+import org.alfresco.util.Pair;
+import org.alfresco.util.ParameterCheck;
+import org.alfresco.util.registry.NamedObjectRegistry;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.dao.ConcurrencyFailureException;
+
+/**
+ * @author Neil Mc Erlean (based on existing webscript controllers in the REST API)
+ * @since 4.0
+ */
+public class BlogServiceImpl implements BlogService
+{
+    public static final String BLOG_COMPONENT = "blog";
+    private static final int MIN_NUMBER_OF_PAGES_FOR_THE_USER_TO_LOOP_THROUGH = 10;
+    
+    /**
+     * The logger
+     */
+    private static Log logger = LogFactory.getLog(BlogServiceImpl.class);
+    
+    // Injected services
+    private NamedObjectRegistry<CannedQueryFactory<BlogPostInfo>> cannedQueryRegistry;
+    private GetBlogPostsCannedQueryFactory draftPostsCannedQueryFactory;
+    private GetBlogPostsCannedQueryFactory publishedPostsCannedQueryFactory;
+    private GetBlogPostsCannedQueryFactory publishedExternallyPostsCannedQueryFactory;

```

Comment:
`publishedExternallyPostsCannedQueryFactory` doesn't appear to be used outside of the getter method.